store/sqlstore/upgrade.go

```diff
@@ -346,6 +346,8 @@ func UpgradeDatabaseToVersion47(sqlStore SqlStore) {
 	if shouldPerformUpgrade(sqlStore, VERSION_4_6_0, VERSION_4_7_0) {
 		sqlStore.AlterColumnTypeIfExists("Users", "Position", "varchar(128)", "varchar(128)")
 		sqlStore.AlterColumnTypeIfExists("OAuthAuthData", "State", "varchar(1024)", "varchar(1024)")
+		sqlStore.RemoveColumnIfExists("ChannelMemberHistory", "Email")
+		sqlStore.RemoveColumnIfExists("ChannelMemberHistory", "Username")
```

Comment:
Is it possible that the `ChannelMemberHistory` has a really high row count on any customer DBs? Just wondering if we should give extra warning to customers about this migration ahead of time if it's going to take a long time to remove these columns

Reply:
It is possible, yes. Large customers will likely have `num_channels*num_users` rows in this table. What's the process for giving customers warning in advance? Does it just need to be mentioned in release notes?

Reply:
I think just let one of the PMs know and they'll get it mentioned in the release notes and maybe directly contact any customers with large deployments

Reply:
Ok. I mentioned it in the ticket and tagged the three PMs.

Reply:
Originally commented on the JIRA ticket, but also posting here.

> ..Why are they deleted? Are they obsolete now, if so, why?

> Also, are the DB column removals to fix a bug? Makes me worried to make database changes this close to the release.

Reply:
I should have been more clear from the start: these columns will **not** be present in the database of any customer that has been using v4.6 or earlier. They were added to our testing databases [in this commit](https://github.com/mattermost/mattermost-server/commit/d3e934d07ac0a58a24a435ea7c5b3bd222ef509a#diff-1830f058e8c1b159f0df87e96e716d8e) on 2018-02-07. This change simply removes the columns if present, and is only here to ensure that unit tests pass as expected, and that the database is fixed for any installation that has previously been running a non-release version of `mattermost-server` (i.e. our test servers). 

We should still notify customers that this is being done as per JW's suggestion, because we make a habit of telling them about database changes, but I don't expect that any customers will actually have to do anything or will be impacted by this change.

Reply:
@MusikPolice Okay, thanks. So the columns never existed in a previous "official" release build of Mattermost. That should be fine then. I'll notify Shift so they're aware but I doubt any customer is affected in this case.

I also won't add it to the release notes.